### PR TITLE
Rr 162738054 sit request edit icon edit

### PR DESCRIPTION
**Pull request by Ryan-Koch**
## Description

Adds the ability to edit existing storage in transit requests in the TSP UI. Adjustments were made to the PATCH end end point in order to accommodate the feature.

## Setup

To get sample data to play with:

```sh
make dev_e2e_populate
```

To run the backend server
```sh
make server_run
```

To run the tsp front end
```sh
make tsp_client_run
```

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](./docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](./docs/backend.md#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](./docs/database.md#zero-downtime-migrations))
  * [ ] Have been communicated to #dp3-engineering
  * [ ] Secure migrations have been tested using `bin/run-prod-migrations`
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers](./docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @ntwyman
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](tbd) for this change
* [this article](tbd) explains more about the approach used.

## Screenshots

SIT Edit 3

### CODE REVIEW

File: src/shared/StorageInTransit/Editor.test.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          submitForm={saveAndClose}
+          formEnabled={false}
+          hasSubmitSucceeded={false}
+        />,
+      );
+    });
+
+    it('renders without crashing', () => {
+      expect(wrapper.exists('div')).toBe(true);
+    });
```
Comment (LeDeep) [inline src/shared/StorageInTransit/Editor.test.jsx]: Testing for the existence of a `div` seems too generic. Testing the class on the div would be better, I think. If the tests checks for `div.storage-in-transit-panel-modal`, then it's clearer what is being tested.

File: src/shared/StorageInTransit/Editor.test.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          formEnabled={true}
+          hasSubmitSucceeded={false}
+        />,
+      );
+    });
+
+    it('renders without crashing', () => {
+      // eslint-disable-next-line
+      expect(wrapper.exists('div')).toBe(true);
+    });
```
Comment (LeDeep) [inline src/shared/StorageInTransit/Editor.test.jsx]: same comment as above

**Comment by LeDeep** [comment]
Nice work! Minor changes to some tests, but otherwise looks great.

File: src/shared/StorageInTransit/Editor.test.jsx (hunk 1 above)
Comment (Ryan-Koch) [reply src/shared/StorageInTransit/Editor.test.jsx]: Makes sense. Changed it to be class specific as you suggested.

File: src/shared/StorageInTransit/Editor.test.jsx (hunk 2 above)
Comment (Ryan-Koch) [reply src/shared/StorageInTransit/Editor.test.jsx]: Same reply as above

**Pull request merged**